client/checkout/classic/upe-deferred-intent-creation/stripe-checkout.js

```diff
@@ -201,21 +203,29 @@ export const checkout = ( api, jQueryForm, paymentMethodType ) => {
 	blockUI( jQueryForm );
 
 	const elements = gatewayUPEComponents[ paymentMethodType ].elements;
-	validateElements( elements, jQueryForm );
-	createStripePaymentMethod( api, elements )
-		.then( ( paymentMethodObject ) => {
-			appendFingerprintInputToForm( jQueryForm, fingerprint );
-			appendPaymentMethodIdToForm(
-				jQueryForm,
-				paymentMethodObject.paymentMethod.id
-			);
-			hasCheckoutCompleted = true;
-			submitForm( jQueryForm );
-		} )
-		.catch( ( error ) => {
-			jQueryForm.removeClass( 'processing' ).unblock();
-			showErrorCheckout( error.message );
-		} );
+
+	( async () => {
+		try {
+			await validateElements( elements, jQueryForm );
+
+			createStripePaymentMethod( api, elements )
+				.then( ( paymentMethodObject ) => {
+					appendFingerprintInputToForm( jQueryForm, fingerprint );
+					appendPaymentMethodIdToForm(
+						jQueryForm,
+						paymentMethodObject.paymentMethod.id
+					);
+					hasCheckoutCompleted = true;
+					submitForm( jQueryForm );
+				} )
+				.catch( ( error ) => {
+					jQueryForm.removeClass( 'processing' ).unblock();
+					showErrorCheckout( error.message );
+				} );
+		} catch ( err ) {
+			return false;
+		}
```

Comment:
```suggestion
		try {
			await validateElements( elements, jQueryForm );

			const paymentMethodObject = await createStripePaymentMethod( api, elements );

			appendFingerprintInputToForm( jQueryForm, fingerprint );
			appendPaymentMethodIdToForm(
				jQueryForm,
				paymentMethodObject.paymentMethod.id
			);
			hasCheckoutCompleted = true;
			submitForm( jQueryForm );
		} catch ( err ) {
			jQueryForm.removeClass( 'processing' ).unblock();
			showErrorCheckout( error.message );
		}
```
I feel like we could consolidate these two `catch` blocks into one, since they both basically do the same thing. If we remove the `showErrorCheckout` call from `validateElements`, we can use this `catch` block to perform that action, since we rethrow the `Error` anyways. This seems a little cleaner to me, but let me know if you think otherwise!

Also, I think the `return false` in this `catch` block is redundant, because the function will always return `false` anyways.

Reply:
Ah, yep this is cleaner indeed. Thanks for the suggestion! Done in https://github.com/Automattic/woocommerce-payments/pull/6262/commits/f5b121e0fe494880dd81f995b31a00ceded88b36.